### PR TITLE
LRAC-14367 | master

### PR DESCRIPTION
**Pull request by marcosapmf**
Jira ticket: https://issues.liferay.com/browse/LRAC-14367
Jira ticket: https://issues.liferay.com/browse/LRAC-14368

### CODE REVIEW

**Comment by ealonso** [comment]
Hey @marcosapmf,

Why if we are only going to have one experiment per page we are returning a list? Could you please update the service.xml to reflect it?

**Comment by marcosapmf** [comment]
Hi @ealonso, thanks for reviewing it. I just forced a push to this pull with the changes you've asked. Can you please take a look at it when you have a time?

Thank you!

**Comment by marcosapmf** [comment]
ci:test:sf

**Comment by marcosapmf** [comment]
ci:test:relevant

File: modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/upgrade/v2_8_1/SegmentsExperimentUpgradeProcess.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+					}
+
+					_segmentsExperimentLocalService.deleteSegmentsExperiment(
+						resultSet.getLong("segmentsExperimentId"));
+				}
+			}
+		}
+	}
+
+	private final SegmentsExperimentLocalService
```
Comment (ealonso) [inline modules/apps/segments/segments-service/src/main/java/com/liferay/segments/internal/upgrade/v2_8_1/SegmentsExperimentUpgradeProcess.java:71]: @marcosapmf We cannot use _segmentsExperimentLocalService inside the segments-service upgrade, we can only use service of other modules.

**Comment by marcosapmf** [comment]
ci:test:sf

**Comment by marcosapmf** [comment]
ci:test:relevant

**Comment by marcosapmf** [comment]
Hi @ealonso, thanks for reviewing it again :smile: 
I just made the required changes and forced a new push. 
Can you take a look at it when you have a chance?
Thank you!

**Comment by ealonso** [comment]
Hey @marcosapmf,

Reviewing your pr, I saw that we have it **historySegmentsExperiments** in **GetDataMVCResourceCommand** where we are getting the history of experimentes, but with your changes we are not going to have more than one experiment since we are forcing it.

Do we want to remove this functionality?

Regards,
Eudaldo.

**Comment by marcosapmf** [comment]
Hi @ealonso! Yes, we used to have a history tab on frontend side which showed  `historySegmentsExperiments`, but we decided to remove that (please see [LRAC-14219](https://liferay.atlassian.net/browse/LRAC-14219)). This property is actually a garbage that remained from the story. I'm gonna remove that in a follow up if it's ok to you :smile: 
Thanks!

**Comment by marcosapmf** [comment]
ci:test:sf

**Comment by marcosapmf** [comment]
ci:test:relevant

**Comment by marcosapmf** [comment]
ci:test:sf

**Comment by marcosapmf** [comment]
ci:test:relevant

**Comment by marcosapmf** [comment]
ci:test:relevant

**Comment by ealonso** [comment]
Please see: https://github.com/liferay-echo/liferay-portal/pull/13359, I didn't include the last two commits, since I start working on it last week. Once we have the first part merge on master, I continue with the other pr and those two commits.

**Pull request closed without merging**